dbos/core.py

```diff
@@ -1,13 +1,51 @@
+import sys
+import traceback
 from concurrent.futures import Future
-from typing import TYPE_CHECKING, TypeVar
+from functools import wraps
```

Comment:
Alternative structure: Put this all in `workflows.py`. Either way works. What do you plan to do with transactions and communicators?

Reply:
Was thinking about putting them in `core` also.